src/routes/api/classes.js

```diff
@@ -11,7 +11,6 @@ router.get('/:index', function(req, res, next) {
 
 router.get('/:index/subclasses', ClassController.showSubclassesForClass);
 router.get('/:index/starting-equipment', ClassController.showStartingEquipmentForClass);
-router.get('/:index/spellcasting', ClassController.showSpellcastingForClass);
```

Comment:
I thought we were still keeping this endpoint but only having it grab from the class object?

Reply:
Yep, yep we were, I...forgot. Let me re-work this a bit, I think it's doable.

Reply:
It's probably going to fail integration tests until we get the other one merged in.

Reply:
This one is going to take a little more work. Separately, I think I'll need to make some tweaks to the integration tests themselves, but yes they will probably fail integration tests in CI until the other one is merged.

Reply:
How strongly do we feel about keeping these spellcasting details accessible via their own endpoint? Asking for 2 reasons:
- it seems fairly straightforward / trivial on the user side to hit the `/classes/{index}` endpoint and pull the `spellcasting` field out from there
- right now there isn't anything accessible via it's own endpoint like this that isn't defined in it's own collection in mongodb, and the db change associated with this PR effectively removes all the documents from the `spellcasting` collection and just drops the raw data into the appropriate document in the `class` collection

I'm not too familiar with mongo/express/mongoose, so I've been doing some reading to try and work out a way to do this, but curious if maybe I'm missing something obvious that y'all could point me toward.

Reply:
I don't feel super strong about it? Just slight backwards compatibility.

Reply:
Same as @bagelbits. I personally am not bothered whether it stays or not, but having that bit of backwards compatibility would be nice, and there is the slight benefit of being able to get a smaller response if all you need is the spellcasting data.

I wouldn't consider "there isn't anything else like this" as a reason not to do it, as long as it has a net positive effect.

As for the obvious way of doing it, I would recommend modifying `classController.js`'s `showSpellcastingForClass` function to something like

```js
Class.findOne({ index: req.params.index })
  .then(class => {
    res.status(200).json(class.spellcasting)
  })
  .catch(err => {
    next(err)
  })
```

Reply:
Also, you can probably reuse the original spellcasting integration tests and unit tests but for that endpoint. So not too much additional work.

Reply:
yeah the tests ended up staying pretty much the same, i added them back in a subsequent commit on this PR for readability's sake, happy to squash the changes back into the commit original commit though